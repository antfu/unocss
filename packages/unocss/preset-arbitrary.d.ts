export * from './dist/preset-arbitrary'
export { default } from './dist/preset-arbitrary'
