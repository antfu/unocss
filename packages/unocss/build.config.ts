import { defineBuildConfig } from 'unbuild'

export default defineBuildConfig([
  {
    name: 'Dual',
    entries: [
      'src/postcss',
    ],
    clean: true,
    declaration: true,
    failOnWarn: false,
    rollup: {
      emitCJS: true,
    },
  },
  {
    name: 'ESM only',
    entries: [
      'src/index',
      'src/webpack',
      'src/vite',
      'src/astro',
      'src/preset-uno',
      'src/preset-icons',
      'src/preset-attributify',
      'src/preset-tagify',
      'src/preset-web-fonts',
      'src/preset-typography',
      'src/preset-wind',
      'src/preset-mini',
    ],
    clean: false,
    declaration: true,
    externals: [
      'vite',
      'astro',
      'webpack',
      '@unocss/webpack',
    ],
  },
])
