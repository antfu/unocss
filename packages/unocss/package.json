{
  "name": "unocss",
  "version": "0.27.0",
  "description": "The instant on-demand Atomic CSS engine.",
  "keywords": [
    "unocss",
    "atomic-css",
    "atomic-css-engine",
    "css",
    "tailwind",
    "windicss"
  ],
  "homepage": "https://github.com/unocss/unocss#readme",
  "bugs": {
    "url": "https://github.com/unocss/unocss/issues"
  },
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/unocss/unocss.git"
  },
  "funding": "https://github.com/sponsors/antfu",
  "author": "Anthony Fu <anthonyfu117@hotmail.com>",
  "sideEffects": false,
  "files": [
    "dist",
    "*.d.ts"
  ],
  "exports": {
    ".": {
      "require": "./dist/index.cjs",
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.ts"
    },
    "./preset-attributify": {
      "require": "./dist/preset-attributify.cjs",
      "import": "./dist/preset-attributify.mjs",
      "types": "./dist/preset-attributify.d.ts"
    },
    "./preset-icons": {
      "require": "./dist/preset-icons.cjs",
      "import": "./dist/preset-icons.mjs",
      "types": "./dist/preset-icons.d.ts"
    },
    "./preset-mini": {
      "require": "./dist/preset-mini.cjs",
      "import": "./dist/preset-mini.mjs",
      "types": "./dist/preset-mini.d.ts"
    },
    "./preset-typography": {
      "require": "./dist/preset-typography.cjs",
      "import": "./dist/preset-typography.mjs",
      "types": "./dist/preset-typography.d.ts"
    },
    "./preset-uno": {
      "require": "./dist/preset-uno.cjs",
      "import": "./dist/preset-uno.mjs",
      "types": "./dist/preset-uno.d.ts"
    },
    "./preset-web-fonts": {
      "require": "./dist/preset-web-fonts.cjs",
      "import": "./dist/preset-web-fonts.mjs",
      "types": "./dist/preset-web-fonts.d.ts"
    },
    "./preset-wind": {
      "require": "./dist/preset-wind.cjs",
      "import": "./dist/preset-wind.mjs",
      "types": "./dist/preset-wind.d.ts"
    },
    "./vite": {
      "require": "./dist/vite.cjs",
      "import": "./dist/vite.mjs",
      "types": "./dist/vite.d.ts"
    }
  },
  "main": "dist/index.cjs",
  "module": "dist/index.mjs",
  "types": "index.d.ts",
  "scripts": {
    "build": "unbuild",
    "stub": "unbuild --stub"
  },
  "dependencies": {
    "@unocss/cli": "workspace:*",
    "@unocss/core": "workspace:*",
    "@unocss/preset-attributify": "workspace:*",
    "@unocss/preset-icons": "workspace:*",
    "@unocss/preset-mini": "workspace:*",
    "@unocss/preset-typography": "workspace:*",
    "@unocss/preset-uno": "workspace:*",
    "@unocss/preset-web-fonts": "workspace:*",
    "@unocss/preset-wind": "workspace:*",
    "@unocss/reset": "workspace:*",
    "@unocss/transformer-directives": "workspace:*",
    "@unocss/transformer-variant-group": "workspace:*",
    "@unocss/vite": "workspace:*"
  },
  "engines": {
    "node": ">=14"
  }
}
