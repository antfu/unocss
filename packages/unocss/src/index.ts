import type { UserConfig } from '@unocss/core'

export * from '@unocss/core'
export { default as presetUno } from '@unocss/preset-uno'
export { default as presetAttributify } from '@unocss/preset-attributify'
export { default as presetIcons } from '@unocss/preset-icons'
export { default as presetWebFonts } from '@unocss/preset-web-fonts'
export { default as presetTypography } from '@unocss/preset-typography'
export { default as presetMini } from '@unocss/preset-mini'
export { default as presetWind } from '@unocss/preset-wind'

export function defineConfig<Theme extends {}>(config: UserConfig<Theme>) {
  return config
}
