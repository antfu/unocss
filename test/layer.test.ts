import { createGenerator } from 'unocss'

describe('layers', () => {
  test('static', async() => {
    const uno = createGenerator({
      rules: [
        ['a', { name: 'bar1' }, { layer: 'a' }],
        ['b', { name: 'bar2' }, { layer: 'b' }],
        [/^c(\d+)$/, ([, d]) => ({ name: d }), { layer: 'c' }],
        [/^d(\d+)$/, ([, d]) => `/* RAW ${d} */`, { layer: 'd' }],
      ],
      shortcuts: [
        ['abc', 'a b c2 d3', { layer: 'shortcuts' }],
      ],
      presets: [],
    })
    const { css } = await uno.generate('a b abc d4 c5')
    expect(css).toMatchSnapshot()
  })
})
