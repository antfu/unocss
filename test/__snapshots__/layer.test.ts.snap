// Vitest Snapshot v1

exports[`layers > static 1`] = `
"/* layer: shortcuts */
.abc{name:bar1;}
.abc{name:bar2;}
.abc{name:4;}
/* layer: a */
.a{name:bar1;}
/* layer: b */
.b{name:bar2;}
/* layer: c */
.c5{name:5;}
/* layer: d */
/* RAW 4 */
/* layer: s */
.abcd{name:bar1;}
.abcd{name:bar2;}
.abcd{name:2;}"
`;
